### DISCUSSION TITLE
hidden namespace alias

### DISCUSSION THREAD

**Discussion by john30** [microsoft/typespec#3530] [Q&A]
is there a way for an emitter to get the context of where a namespace was aliased?
I'd like to do something like this:

having a file `import.tsp` to be imported from another one:
```typespec
namespace importns;
model Foo {
  value: integer,
}
```
and the main file:
```typespec
import "./import.ts";
namespace mainns;
alias usens = importns;
```
With that I'd like an emitter to know that all models of the `usens` are part of its namespace as well.
I guess this would potentially be a `namespaceLiteralContext` in the emitter framework, but it seems not to exist, or at least I couldn't find it

**Comment by timotheeguerin** [comment]
aliases in TypeSpec are gone in the type graph and should be seen as a way to reuse some piece of code around not as a way to redeclare types.

Could you share more about what you are trying to achieve/produce?

**Reply by john30** [reply]
basically I try to realize an "import all models" from another tsp file in order to add extend one namespace with those, so that these shared models basically can appear within several namespaces
